be/src/exec/vectorized/sorting/sort_permute.h

```diff
@@ -0,0 +1,48 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <vector>
+
+#include "column/vectorized_fwd.h"
+#include "simd/simd.h"
+
+namespace starrocks::vectorized {
+
+struct PermutationItem {
+    uint32_t chunk_index;
+    uint32_t index_in_chunk;
+    uint32_t permutation_index;
+};
+
+struct SmallPermuteItem {
+    uint32_t index_in_chunk;
+
+    bool operator==(const SmallPermuteItem& rhs) const { return index_in_chunk == rhs.index_in_chunk; }
+};
+
+// Inline data value into the permutation to optimize cache efficiency
+template <class T>
+struct InlinePermuteItem {
+    // NOTE: do not inline a large value
+    static_assert(sizeof(T) <= 16, "Do not inline a large value");
+
+    T inline_value;
+    uint32_t index_in_chunk;
+};
+
+using Tie = std::vector<uint8_t>;
```

Comment:
It's better to add comment indicate that consecutive '1' means equal range.

Reply:
Good suggestion. I will write some explanation about it.